rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java

```diff
@@ -693,6 +706,157 @@ void getOldFederationBtcUTXOs_calledTwice_returnCachedUtxos() {
 
     }
 
+    @Test
+    void getNewFederation_initialVersion() {
+        List<Integer> storageCalls = new ArrayList<>();
+        List<Integer> deserializeCalls = new ArrayList<>();
+        Federation newFederation = buildMockFederation(100, 200, 300);
+        Repository repositoryMock = mock(Repository.class);
```

Comment:
Let's use the InMemoryStorage, will simplify this a lot

Reply:
Agree with Marcos